docs/getting-started/how-prisma-schemas-are-converted-into-entities.md

```diff
@@ -0,0 +1,283 @@
+
+Amplication currently does not support adding custom attributes to enum and enum keys.
+Instead, Amplication instead allows you to define options as an `optionSet` or `multiSelectOptionSet`.
+
+So if your Prisma schema includes `@map` or `@@map` custom attributes on an enum or enum keys, these attributes will not be accurately represented after processing.
+The final Prisma schema file will not contain these mapping attributes.  
+
+If your app logic relies on `@map` and `@@map` custom attributes, you'll need to manually adjust the generated `schema.prisma` file or your app code to ensure everything works correctly.
+
+Here's how that log might look:
+
+```bash title="Amplication Upload Log" 
+2023-07-06T09:47:39.410Z (Warning) The option 'PENDING' has been created in the enum 'OrderStatus', but its value has not been mapped
+2023-07-06T09:47:39.410Z (Info) The option 'COMPLETED' has been created in the enum 'OrderStatus'
+2023-07-06T09:47:39.410Z (Info) The option 'CANCELLED' has been created in the enum 'OrderStatus'
+2023-07-06T09:47:39.411Z (Warning) The option 'INDIVIDUAL' has been created in the enum 'CustomerType', but its value has not been mapped  
+2023-07-06T09:47:39.411Z (Warning) The option 'COMPANY' has been created in the enum 'CustomerType', but its value has not been mapped
+```
+
+### 2. Composite IDs on models   
+
+A model in Prisma can use the `@@id` custom attribute and combine fields to create a unique identifier for each record.
+But, due to current limitations, Amplication requires that each model have a single, distinct `id` field.   
+
+During schema processing. Amplication modifies IDs created with the `@@id` custom attribute to be `@@unique` custom attributes, which are supported.
+It also introduces a new `id` field of type `String` to each model.   
+
+For example, if your generated `schema.prisma` file looks like the following:
+
+```tsx title="schema.prisma"
+model Employee {
+  firstName String
+  lastName  String
+  birthdate DateTime
+
+  @@id([firstName, lastName, birthdate])
+}
+```
+
+Amplication will transform it to look like the following:  
+
+```diff title="schema.prisma"
+model Employee {
++ id String @id @default(cuid())
+  firstName String
+  lastName  String
+  birthdate DateTime
+
+- @@id([firstName, lastName, birthdate])  
++ @@unique([firstName, lastName, birthdate])
+}
+```
+
+As the schema is processed, we log warning entries if we encounter a model without an `id` field and an existing `@@id` attribute.
+
+Here's an example of how that log might look:
+
+```bash title="Amplication Upload Log"
+2023-07-08T08:03:00.559Z (Warning) Attribute "id" was changed to "unique" on model "Employee"
+2023-07-08T08:03:00.559Z (Warning) id field was added to model "Employee" 
+```
+
+This approach this maintains uniqueness and integrity for each record in the database, but it may impact how relationships and queries are designed, given the shift from a composite id with `@@id` to a singular one.
```

Comment:
```suggestion
This approach maintains uniqueness and integrity for each record in the database, but it may impact how relationships and queries are designed, given the shift from a composite id with `@@id` to a singular one.
```